deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/CustomLayer/ElementWiseMultiplicationLayer_impl.java

```diff
@@ -0,0 +1,160 @@
+        INDArray wErrorSignal = errorSignal.mul(W);
+        nextLayerGradient.gradientForVariable().put(ElementWiseParamInitializer.WEIGHT_KEY, wErrorSignal);
+        return nextLayerGradient;
+    }
+
+    @Override
+    public Gradient calcGradient(Gradient layerError, INDArray activation) {
+        Gradient ret = new DefaultGradient();
+        INDArray weightErrorSignal = layerError.getGradientFor(ElementWiseParamInitializer.WEIGHT_KEY);
+        INDArray weightError = weightErrorSignal.mul(activation);
+        ret.gradientForVariable().put(ElementWiseParamInitializer.WEIGHT_KEY, weightError);
+        INDArray biasGradient = weightError.mean(0);
+        ret.gradientForVariable().put(ElementWiseParamInitializer.BIAS_KEY, biasGradient);
+
+        return ret;
+    }
+
+    @Override
+    public Pair<Gradient, INDArray> backpropGradient(INDArray epsilon) {
+        //If this layer is layer L, then epsilon for this layer is ((w^(L+1)*(delta^(L+1))^T))^T (or equivalent)
+        INDArray z = preOutput(true); //Note: using preOutput(INDArray) can't be used as this does a setInput(input) and resets the 'appliedDropout' flag
+        //INDArray activationDerivative = Nd4j.getExecutioner().execAndReturn(Nd4j.getOpFactory().createTransform(conf().getLayer().getActivationFunction(), z).derivative());
+        //        INDArray activationDerivative = conf().getLayer().getActivationFn().getGradient(z);
+        //        INDArray delta = epsilon.muli(activationDerivative);
+        INDArray delta = layerConf().getActivationFn().backprop(z, epsilon).getFirst(); //TODO handle activation function params
+//        if activation is a identity function
+//        delta = Nd4j.ones(this.layerConf().getNIn());
+
+        if (maskArray != null) {
+            applyMask(delta);
+        }
+
+        Gradient ret = new DefaultGradient();
+
+//        INDArray weightGrad = Nd4j.zeros( gradientViews.get(ElementWiseParamInitializer.WEIGHT_KEY).shape(),'f'); //f order
+        INDArray weightGrad =  gradientViews.get(ElementWiseParamInitializer.WEIGHT_KEY); //f order
+//        reset weight gradients
+        weightGrad.subi(weightGrad);
+
+        for(int row =0;row<input.rows();row++){
+            weightGrad.addi(input.getRow(row).mul(delta.getRow(row)));
+        }
+
+//        weightGrad.addi(input.getRow(0).mul(delta));
+
+        INDArray biasGrad = gradientViews.get(ElementWiseParamInitializer.BIAS_KEY);
+        delta.sum(biasGrad, 0); //biasGrad is initialized/zeroed first
+
+        ret.gradientForVariable().put(ElementWiseParamInitializer.WEIGHT_KEY, weightGrad);
+        ret.gradientForVariable().put(ElementWiseParamInitializer.BIAS_KEY, biasGrad);
+
+        INDArray epsilonNext = Nd4j.zeros(params.get(ElementWiseParamInitializer.WEIGHT_KEY).shape());
+
+        for(int row =0;row<delta.rows();row++){
+            epsilonNext.addi(params.get(ElementWiseParamInitializer.WEIGHT_KEY).mul(delta.getRow(row)));
```

Comment:
This for loop is inefficient.
I believe it can be replaced by:
```epsilonNext = delta.mulRowVector(params.get(ElementWiseParamInitializer.WEIGHT_KEY).sum(0)```